web/client/components/TOC/enhancers/tocItemsSettings.js

```diff
@@ -0,0 +1,104 @@
+            Object.keys(newParams).forEach((key) => {
+                originalSettings[key] = props.initialSettings[key];
+            });
+            props.onUpdateOriginalSettings(originalSettings);
+            props.onUpdateSettings(newParams);
+            if (update) {
+                props.onUpdateNode(
+                    props.settings.node,
+                    props.settings.nodeType,
+                    {...props.settings.options, ...newParams}
+                );
+            }
+        },
+        onClose: ({onUpdateNode, originalSettings, settings, onHideSettings, onShowAlertModal}) => forceClose => {
+            const originalOptions = Object.keys(settings.options).reduce((options, key) => ({...options, [key]: key === 'opacity' && !originalSettings[key] && 1.0 || originalSettings[key]}), {});
+            if (!isEqual(originalOptions, settings.options) && !forceClose) {
+                onShowAlertModal(true);
+            } else {
+                onUpdateNode(
+                    settings.node,
+                    settings.nodeType,
+                    {...settings.options, ...originalSettings}
+                );
+                onHideSettings();
+                onShowAlertModal(false);
+            }
+        },
+        onSave: ({onHideSettings = () => {}, onShowAlertModal = () => {}}) => () => {
+            onHideSettings();
+            onShowAlertModal(false);
+        }
+    }),
+    lifecycle({
+        componentWillMount() {
+            const {
+                element = {},
+                onUpdateOriginalSettings = () => {},
+                onUpdateInitialSettings = () => {}
+            } = this.props;
+
+            onUpdateOriginalSettings({...element});
+            onUpdateInitialSettings({...element});
+        },
+        componentWillReceiveProps(newProps) {
+            // an empty description does not trigger the single layer getCapabilites,
+            // it does only for missing description
+            const {
+                settings = {},
+                onRetrieveLayerData = () => {}
+            } = this.props;
+
+            if (!settings.expanded && newProps.settings && newProps.settings.expanded && isNil(newProps.element.description) && newProps.element.type === "wms") {
+                onRetrieveLayerData(newProps.element);
+            }
+        },
+        componentWillUpdate(newProps) {
+            const {
+                settings = {},
+                onUpdateOriginalSettings = () => {},
+                onUpdateInitialSettings = () => {},
+                onSetTab = () => {}
+            } = this.props;
+
+            if (!settings.expanded && newProps.settings && newProps.settings.expanded) {
+                // update initial and original settings
+                onUpdateOriginalSettings({...newProps.element});
+                onUpdateInitialSettings({...newProps.element});
+                onSetTab('general');
+            }
+        }
+    })
+);
+
```

Comment:
Please, describe what you get applying this enhancer to the TOCItemsSettings component